program.py

```diff
@@ -7,111 +7,73 @@
 # Import necessary packages
 # ************************************************************************
 
-import configparser
 import json
 import time
 import datetime
-import platform
-import socket
 import gzip
 import random
 import requests
 import traceback
-import base64
-
+import os
+from urllib.parse import urlparse
 
 # ************************************************************************
-# Specify options for sending web requests to the target PI System
+# Global Variables
 # ************************************************************************
 
-# Whether or not we are inferring where the app is pointing to from the config file
-forceSending = False
-
-# Specifys whether we are sending to PI or OCS.  The main changes are in the accepted messages and the URL.
-sendingToOCS = True
-
-# Specify whether to compress OMF message before
-# sending it to ingress endpoint
-USE_COMPRESSION = False
-
-# Set this to the path of the certificate pem file if you using a self signed cert.
-# Set this to True if your cert is trusted by the Python certify.
-# Set to False if you do not want to check the certificate (NOT RECOMMENDED)
-VERIFY_SSL = True
-
-# Specify the timeout, in seconds, for sending web requests
-# (if it takes longer than this to send a message, an error will be thrown)
-WEB_REQUEST_TIMEOUT_SECONDS = 30
-
-# Holder for the producer token.  It is set from the configuration
-producerToken = ""
-
-# Holder for the omfEndPoint if sending to PI.  It is set from the configuration
-omfEndPoint = ""
-
-# Holder for the checkbase if sending to PI.  It is set from the configuration
-checkBase = ""
-
-# Holder for the omfEndPoint base if sending to OCS.  Auth and OMF endpoint are built from this.  It is set from the configuration
-resourceBase = ""
-
-dataServerName = ""
-
-username = ""
-
-password = ""
-
-
 # The version of the OMFmessages
 omfVersion = "1.1"
 
-# Holders for data message values
-integer_boolean_value = 0
-string_boolean_value = "True"
-integer_index1 = 0
-integer_index2_1 = 1
-integer_index2_2 = 1
+# The configurations of the endpoints to send to
+endpoints = None
 
-# Token information
-__expiration = 0
-__token = ""
+# List of possible endpoint types
+endpoint_types = ["OCS", "EDS", "PI"]
```

Comment:
Seems like it would be better to do this as an `Enum` instead of matching specific indices?

Reply:
Agreed. Changed to string Enum